homeassistant/components/simplepush/__init__.py

```diff
@@ -1 +1,42 @@
 """The simplepush component."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DATA_HASS_CONFIG, DOMAIN
+
+PLATFORMS = [Platform.NOTIFY]
+
+CONFIG_SCHEMA = cv.removed(DOMAIN, raise_if_present=False)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the simplepush component."""
+
+    hass.data[DATA_HASS_CONFIG] = config
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up simplepush from a config entry."""
+
+    hass.async_create_task(
+        discovery.async_load_platform(
+            hass,
+            Platform.NOTIFY,
+            DOMAIN,
+            dict(entry.data),
+            hass.data[DATA_HASS_CONFIG],
```

Comment:
How does this work exactly, why do we need to pass in the yaml config when configuring a config entry?

Reply:
Maybe it was actually needed in the tibber integration, whereas it's just ignored here?

Reply:
The same has been discussed in `nfandroidtv` ( #73227 ) integration and highlighted by @MartinHjelmare. Martin can you please confirm if this is required.

Reply:
Yes, `discovery.async_load_platform` requires the original config as fifth parameter. That's the signature and interface.